### PR TITLE
Vassal fold ensures you're a revenge vassal

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Closes https://github.com/fulpstation/fulpstation/pull/1195

Fixes ex-vassal ability because I did it stupidly and was checking for the wrong antag type and was early returning if they HAD said antag type AGhHH im stupid

I also threw another bloodsucker fix, constant cost powers now properly checks if you can afford the power if you're not a bloodsucker (vassal stuff)

### CODE REVIEW

File: fulp_modules/features/antagonists/bloodsuckers/code/powers/vassal/vassal_fold.dm [hunk 1]
```diff
@@ -22,12 +22,13 @@
 	. = ..()
 	if(!.)
 		return FALSE
-	var/datum/antagonist/vassal/revenge/revenge_vassal = owner.mind.has_antag_datum(/datum/antagonist/ex_vassal)
-	if(revenge_vassal)
+	var/datum/antagonist/vassal/revenge/revenge_vassal = owner.mind.has_antag_datum(/datum/antagonist/vassal/revenge)
+	if(!revenge_vassal)
```
Comment (Bilbo367) [inline fulp_modules/features/antagonists/bloodsuckers/code/powers/vassal/vassal_fold.dm:26]:
```suggestion
	if(isnull(revenge_vassal))
```

Comment (remove32) [reply fulp_modules/features/antagonists/bloodsuckers/code/powers/vassal/vassal_fold.dm:26]: arent nulls in DM considered falsey? i dont know what difference this makes aside from explanatory power

Comment (Bilbo367) [reply fulp_modules/features/antagonists/bloodsuckers/code/powers/vassal/vassal_fold.dm:26]: readability and performance its a nit, not anything that would realistically change anything.

Comment (JohnFulpWillard) [reply fulp_modules/features/antagonists/bloodsuckers/code/powers/vassal/vassal_fold.dm:26]: It don't matteeeeeeeeerrr

**Pull request merged**